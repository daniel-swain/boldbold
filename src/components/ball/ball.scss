$baseHeight: 0px;
$baseWidth: 0px;

.ball {
  height: $baseHeight;
  width: $baseWidth;
  border-radius: 100%;
  position: absolute;

  &.sm {
    height: $baseHeight;
    width: $baseWidth;
  }

  &.md {
    height: $baseHeight * 3;
    width: $baseWidth * 3;
  }

  &.lg {
    height: $baseHeight * 6;
    width: $baseWidth * 6;
  }

  &.xl {
    height: $baseHeight * 9;
    width: $baseWidth * 9;
  }
}

.ball {
  &.primary {
    background-color: $primary;
  }
  &.accent {
    background-color: $accent;
  }
  &.red {
    background: red;
  }
  &.green {
    background: green;
  }
  &.blue {
    background: blue;
  }
}
