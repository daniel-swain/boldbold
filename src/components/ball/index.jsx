import React from 'react';
import PropTypes from 'prop-types';

const Ball = ({ className }) => <div className={`${className} ball`} />;

Ball.defaultProps = {
  className: ''
};

Ball.propTypes = {
  className: PropTypes.string
};

export default Ball;
