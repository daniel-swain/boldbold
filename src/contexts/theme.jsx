import React, { createContext, Component } from 'react';
import PropTypes from 'prop-types';

const DEFAULT_THEME = 'default';
const ALTERNATE_THEME = 'alternate';

const PINK = '#ff80ab';
const LIGHT_PINK = '#FCE4EC';
const WHITE = '#fff';

const themes = {
  [DEFAULT_THEME]: {
    primary: PINK,
    accent: LIGHT_PINK
  },
  [ALTERNATE_THEME]: {
    primary: LIGHT_PINK,
    accent: WHITE
  }
};

export const ThemeContext = createContext({
  current: DEFAULT_THEME,
  theme: themes[DEFAULT_THEME]
});

class Theme extends Component {
  static propTypes = { children: PropTypes.node.isRequired }

  state = { current: DEFAULT_THEME, theme: themes[DEFAULT_THEME] }

  render() {
    return (
      <ThemeContext.Provider value={{ ...this.state }}>
        {this.props.children}
      </ThemeContext.Provider>
    );
  }
}

export const withTheme = WrappedComponent => props => (
  <ThemeContext.Consumer>
    {context =>
      <WrappedComponent {...props} theme={context.theme} />
    }
  </ThemeContext.Consumer>
);


export default Theme;
