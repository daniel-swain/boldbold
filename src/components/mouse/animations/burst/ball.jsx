import styled from 'styled-components';
import { withTheme } from '../../../contexts/theme';

export const Ball = styled.div`
  height: 0px;
  width: 0px;
  border-radius: 100%;
  position: absolute;

  &.primary {
    background-color: ${props => props.theme.primary};
  }

  &.accent {
    background-color: ${props => props.theme.accent};
  }
`;

Ball.defaultProps = {
  theme: {
    primary: 'white',
    accent: 'black'
  }
};

export default withTheme(Ball);
// export default Ball;
