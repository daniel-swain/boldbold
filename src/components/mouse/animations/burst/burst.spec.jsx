/* global beforeAll, describe, it, expect, shallow */
import React from 'react';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import { themes, DEFAULT_THEME } from '../../../../contexts/theme';
import Burst from './index';
import { Ball } from './ball';

describe('<Burst />', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Burst x={0} y={0} />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('<Ball />', () => {
    it('should render', () => {
      const ballJson = toJson(shallow(<Ball theme={themes[DEFAULT_THEME]} />));
      expect(ballJson).toMatchSnapshot();
    });
  });
});
