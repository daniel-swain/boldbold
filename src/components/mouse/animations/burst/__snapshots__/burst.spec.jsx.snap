// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Burst /> <Ball /> should render 1`] = `
.c0 {
  height: 0px;
  width: 0px;
  border-radius: 100%;
  position: absolute;
}

.c0.primary {
  background-color: #ff80ab;
}

.c0.accent {
  background-color: #FCE4EC;
}

<div
  className="c0"
/>
`;

exports[`<Burst /> should render 1`] = `
Array [
  <Component
    className="primary ball-0 ball"
    key="primary0"
    style={
      Object {
        "zIndex": 0,
      }
    }
  />,
  <Component
    className="accent ball-1 ball"
    key="accent1"
    style={
      Object {
        "zIndex": 1,
      }
    }
  />,
  <Component
    className="primary ball-2 ball"
    key="primary2"
    style={
      Object {
        "zIndex": 2,
      }
    }
  />,
  <Component
    className="accent ball-3 ball"
    key="accent3"
    style={
      Object {
        "zIndex": 3,
      }
    }
  />,
  <Component
    className="primary ball-4 ball"
    key="primary4"
    style={
      Object {
        "zIndex": 4,
      }
    }
  />,
  <Component
    className="accent ball-5 ball"
    key="accent5"
    style={
      Object {
        "zIndex": 5,
      }
    }
  />,
  <Component
    className="primary ball-6 ball"
    key="primary6"
    style={
      Object {
        "zIndex": 6,
      }
    }
  />,
  <Component
    className="accent ball-7 ball"
    key="accent7"
    style={
      Object {
        "zIndex": 7,
      }
    }
  />,
  <Component
    className="primary ball-8 ball"
    key="primary8"
    style={
      Object {
        "zIndex": 8,
      }
    }
  />,
  <Component
    className="accent ball-9 ball"
    key="accent9"
    style={
      Object {
        "zIndex": 9,
      }
    }
  />,
  <Component
    className="primary ball-10 ball"
    key="primary10"
    style={
      Object {
        "zIndex": 10,
      }
    }
  />,
  <Component
    className="accent ball-11 ball"
    key="accent11"
    style={
      Object {
        "zIndex": 11,
      }
    }
  />,
  <Component
    className="primary ball-12 ball"
    key="primary12"
    style={
      Object {
        "zIndex": 12,
      }
    }
  />,
  <Component
    className="accent ball-13 ball"
    key="accent13"
    style={
      Object {
        "zIndex": 13,
      }
    }
  />,
  <Component
    className="primary ball-14 ball"
    key="primary14"
    style={
      Object {
        "zIndex": 14,
      }
    }
  />,
  <Component
    className="accent ball-15 ball"
    key="accent15"
    style={
      Object {
        "zIndex": 15,
      }
    }
  />,
  <Component
    className="primary ball-16 ball"
    key="primary16"
    style={
      Object {
        "zIndex": 16,
      }
    }
  />,
  <Component
    className="accent ball-17 ball"
    key="accent17"
    style={
      Object {
        "zIndex": 17,
      }
    }
  />,
  <Component
    className="primary ball-18 ball"
    key="primary18"
    style={
      Object {
        "zIndex": 18,
      }
    }
  />,
  <Component
    className="accent ball-19 ball"
    key="accent19"
    style={
      Object {
        "zIndex": 19,
      }
    }
  />,
]
`;
